src/client.rs

```diff
@@ -781,6 +768,70 @@ where
     }
 }
 
+#[cfg(feature = "surf")]
+impl<F> Client<F>
+where
+    F: EnumArray<CachedFeature> + Debug + DeserializeOwned + Serialize,
+{
+    async fn get_features(&self, endpoint: &str) -> Result<Features, Error> {
+        self.http.get(&endpoint).recv_json::<Features>().await
+    }
+
+    async fn upload_metrics(&self, endpoint: &str, metrics: &Metrics) -> bool {
+        let req = self.http.post(endpoint);
+        if let Ok(req) = req.body_json(metrics) {
+            if let Ok(res) = req.await {
+                if res.status().is_success() {
+                    return true;
+                }
+            }
+        }
+
+        false
+    }
+
+    async fn do_registration(&self, registration: &Registration) -> Result<Response, Error> {
+        self.http
+            .post(&Registration::endpoint(&self.api_url))
+            .body_json(&registration)?
+            .await
+    }
+}
+
+#[cfg(feature = "reqwest")]
```

Comment:
I think these blocks might be better with `cfg_if!` or so - it would prevent double or zero defining in future and generally make it less error prone.